xtask/src/main.rs

```diff
@@ -0,0 +1,134 @@
+use std::{fs::File, io::Write, path::PathBuf, process::Command};
+
+use anyhow::{anyhow, Result};
+use bindgen::Builder;
+use directories::UserDirs;
+use log::LevelFilter;
+
+#[derive(Debug, PartialEq)]
+enum Arch {
+    RiscV,
+    Xtensa,
+}
+
+fn main() -> Result<()> {
+    env_logger::Builder::new()
+        .filter_module("xtask", LevelFilter::Info)
+        .init();
+
+    // The directory containing the cargo manifest for the 'xtask' package is a
+    // subdirectory within the cargo workspace:
+    let workspace = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+    let workspace = workspace.parent().unwrap().canonicalize()?;
+
+    // Determine the $HOME directory, and subsequently the Espressif tools
+    // directory:
+    let home = UserDirs::new().unwrap().home_dir().to_path_buf();
+    let tools = home.join(".espressif").join("tools");
+
+    generate_bindings_for_chip(
+        "esp32",
+        Arch::Xtensa,
+        &workspace,
+        tools.join(
+            "xtensa-esp32-elf/esp-2021r2-patch5-8_4_0/xtensa-esp32-elf/xtensa-esp32-elf/include/",
```

Comment:
Depending on how these tools are installed, they could be different. What about finding them from $PATH? Activating an esp-idf env will put them there, and I imagine only maintainers will use this xtask?

Reply:
Yeah this was just copied from the existing `mkbindings.bat` script, I'm sure there are definitely lots of improvements to be made.

Reply:
@bjoernQ did you want this changed in this PR, or can I handle it in a separate one? Given that only you were able to generate the bindings prior to this, I don't think it's too much of an issue right now 😁 It does at least work with the current toolchain.

Reply:
I'm totally fine with changing that in a separate PR